[BUG] - char gen drops out on character creation with hard error

### Bug Checklist

- [x] I have searched for existing issues to see if this has already been reported
- [x] The issue occurs while all Modules are disabled

### What Happened?

selecting a disciplin kills it.

foundry.mjs:30113 Uncaught (in promise) Error: Failed to render Application "character-generation-prompt-102":
Failed to render template part "classes":
Cannot read properties of undefined (reading 'typeLabels')
    at #render (foundry.mjs:30113:13)
    at async #try (foundry.mjs:7550:17)Caused by: Error: Failed to render template part "classes":
Cannot read properties of undefined (reading 'typeLabels')
    at CharacterGenerationPrompt._renderHTML (foundry.mjs:31838:17)
    at async #render (foundry.mjs:30104:22)
    at async #try (foundry.mjs:7550:17)Caused by: TypeError: Cannot read properties of undefined (reading 'typeLabels')
    at String.linkForUuidSync (formatting.mjs:56:29)
    at String.c (handlebars.min.js:27:27672)
    at eval (eval at createFunctionContext (handlebars.min.js:29:10783), <anonymous>:13:145)
    at h (handlebars.min.js:27:25711)
    at c (handlebars.min.js:27:6945)
    at Array.<anonymous> (handlebars.min.js:27:7335)
    at Array.c (handlebars.min.js:27:27672)
    at eval (eval at createFunctionContext (handlebars.min.js:29:10783), <anonymous>:11:49)
    at h (handlebars.min.js:27:25711)
    at c (handlebars.min.js:27:6945)

### What Did You Expect to Happen?

select a disciplin during char gen sould work

### Steps to Reproduce

select a disciplin during char gen

### What core and system version are you reporting this for?

v14

### What browser are you using?

FoundryVTT Desktop App

### Anything Else?

_No response_

--- a/module/helpers/formatting.mjs
+++ b/module/helpers/formatting.mjs
@@ -51,16 +51,16 @@ export function linkForUuidSync( uuid ) {
   const parsedUuid = foundry.utils.parseUuid( uuid );
   const doc = fromUuidSync( uuid, { strict: false } );
   const name = doc?.name ?? "";
-  const packId = parsedUuid.collection?.metadata?.id ?? "";
+  const packId = parsedUuid?.collection?.metadata?.id ?? "";
   const tooltipType = game.i18n.localize(
-    CONFIG[parsedUuid.type].typeLabels[doc?.type]
+    CONFIG[parsedUuid?.type]?.typeLabels[doc?.type]
   );
 
   if ( !doc ) return `
     <a
       class="content-link broken"
       data-uuid="${ uuid } "
-      data-type="${ parsedUuid.type }"
+      data-type="${ parsedUuid?.type }"
       data-tooltip="${ tooltipType }"
       data-pack="${ packId }"
     >

--- a/templates/global/card-options-effect.hbs
+++ b/templates/global/card-options-effect.hbs
@@ -1,11 +1,11 @@
 <div class="itemControlGroup" data-item-id="{{_id}}">
-  <a class="item-control item-display" data-action="displayChild" data-uuid="{{uuid}}" title="{{localize "ED.Controls.displayItem"}}">
+  <a class="item-control item-display" data-action="displayChild" data-uuid="{{_uuid}}" title="{{localize "ED.Controls.displayItem"}}">
     <i class="fas fa-comments"></i>
   </a>
-  <a class="item-control effect-delete" data-action="deleteChild" data-uuid="{{uuid}}" title="{{localize "ED.Controls.delete"}}">
+  <a class="item-control effect-delete" data-action="deleteChild" data-uuid="{{_uuid}}" title="{{localize "ED.Controls.delete"}}">
     <i class="fas fa-trash"></i>
   </a>
-  <a class="item-control effect-edit" data-action="editChild" data-uuid="{{uuid}}" title="{{localize "ED.Controls.edit"}}">
+  <a class="item-control effect-edit" data-action="editChild" data-uuid="{{_uuid}}" title="{{localize "ED.Controls.edit"}}">
     <i class="fas fa-edit"></i>
   </a>
 </div>

--- a/templates/item/item-partials/item-details/item-effects.hbs
+++ b/templates/item/item-partials/item-details/item-effects.hbs
@@ -19,7 +19,7 @@
 
     <div>
       {{#each item.effects as |effect id|}}
-        {{> "systems/ed4e/templates/global/effect-card.hbs" effect }}
+        {{> "systems/ed4e/templates/global/effect-card.hbs" effect isItemEffect=true}}
       {{/each}}
     </div>
 

